### PR TITLE
Add logging capability

### LINKED ISSUE #140: Add logging capability
I don't think we need to go overboard with logging, but it would be useful to log when connections and reconnections are occurring, and why. Once we have Sentinel support we'd also want to log the outcomes of Sentinel discovery.

It might also help users debugging apps if there was an option to log a trace of commands and responses?

Probably [zio-logging](https://github.com/zio/zio-logging) is the way to go.

### PR DESCRIPTION
**Pull request by anovakovic01**
This PR resolves #140.

**Summary:**
1. Add logs for connecting, reconnecting, and disconnecting operations,
2. Add debug logs for sending commands and receiving results.

### CODE REVIEW

File: redis/src/main/scala/zio/redis/Interpreter.scala [hunk 1]
```diff
@@ -29,13 +31,20 @@ trait Interpreter {
...
-      byteStream: Managed[IOException, ByteStream.ReadWriteBytes]
+      byteStream: Managed[IOException, ByteStream.ReadWriteBytes],
+      logger: Logger[String]
     ) extends Service {
 
       override def execute(command: Chunk[RespValue.BulkString]): IO[RedisError, RespValue] =
-        Promise
-          .make[RedisError, RespValue]
-          .flatMap(promise => reqQueue.offer(Request(command, promise)) *> promise.await)
+        logger.locally(LogAnnotation.CorrelationId(Some(UUID.randomUUID()))) {
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Interpreter.scala:39]: I wouldn't do any logging here.

File: redis/src/main/scala/zio/redis/Interpreter.scala [hunk 2]
```diff
@@ -66,30 +75,34 @@ trait Interpreter {
             sendWith(rwBytes.write).forever race runReceive(rwBytes.read)
           }
           .tapError { e =>
-            // TODO - log that a reconnection is happening
-            resQueue.takeAll.flatMap(IO.foreach_(_)(_.fail(e)))
+            logger.warn(s"Reconnecting due to error: $e") *>
+              resQueue.takeAll.flatMap(IO.foreach_(_)(_.fail(e)))
           }
           .retryWhile(Function.const(true))
+          .tapError { e =>
```
Comment (mijicd) [inline redis/src/main/scala/zio/redis/Interpreter.scala:82]: You can drop the block syntax here.

**Review by mijicd — COMMENTED** [review]
Make sure to reformat the sources.

File: redis/src/main/scala/zio/redis/Interpreter.scala (hunk 1 above)
Comment (anovakovic01) [reply redis/src/main/scala/zio/redis/Interpreter.scala]: Removed.

File: redis/src/main/scala/zio/redis/Interpreter.scala (hunk 2 above)
Comment (anovakovic01) [reply redis/src/main/scala/zio/redis/Interpreter.scala]: Done.

**Review by mijicd — APPROVED** [review]

**Pull request merged**